packages/@aws-cdk/aws-route53-targets/test/cloudfront-target.test.ts

```diff
@@ -39,20 +34,17 @@ test('use CloudFrontTarget hosted zone id mappings in nested stacks', () => {
   targets.CloudFrontTarget.getHostedZoneId(nestedStackB);
 
   // THEN
-  for (let nestedStack of [nestedStackA, nestedStackB]) {
-    expect(SynthUtils.toCloudFormation(nestedStack)).toEqual({
-      Mappings: {
-        AWSCloudFrontPartitionHostedZoneIdMap: {
-          'aws': {
-            zoneId: 'Z2FDTNDATAQYW2',
-          },
-          'aws-cn': {
-            zoneId: 'Z3RFFRIM2A3IF5',
-          },
-        },
-      },
-    });
-  }
+  // for (let nestedStack of [nestedStackA, nestedStackB]) {
+  //   console.log(nestedStackA.artifactId, nestedStackA.stackName);
+  //   Template.fromStack(nestedStack).hasMapping('AWSCloudFrontPartitionHostedZoneIdMap', {
```

Comment:
This is a potential bug that I need help diagnosing and/or providing a workaround.

```ts
import { NestedStack} from '@aws-cdk/core';

const nestedStack = new NestedStack(stack, 'ns');
Template.fromStack(nestedStack)...
```

Fails with `unable to find artifact with id "undefined"`. I can trace this to the `fromStack()` method, which calls `toTemplate()`, which returns `assembly.getStackArtifact(stack.artifactId).template`. The failure is because `nestedStack.artifactId == undefined`.

Reply:
I suspect this is because 'assertions' does not support nested stacks. We should add that support here.

Something like this - https://github.com/aws/aws-cdk/blob/6390cb581ca924d0aa7114a79b877ae1afd7b96a/packages/%40aws-cdk/assert-internal/lib/synth-utils.ts#L55-L58